var/spack/repos/builtin/packages/cbtf-krell/package.py

```diff
@@ -104,6 +100,8 @@ class CbtfKrell(Package):
 
```

Comment:
Above this line you have variants and dependencies for every possible MPI provider. This isn't the way to do it. You should use:
```python
variant('mpi', default=True, description='Build with MPI support')

depends_on('mpi', when='+mpi')
```
This uses the virtual dependency. With this method, it becomes impossible to build with multiple MPI dependencies. You can select which MPI package is used on the command line like so:
```
$ spack install cbtf-krell +mpi ^openmpi
```